.travis.yml

```diff
@@ -0,0 +1,51 @@
+language: ruby
+
+env:
+  global:
+    - DISCOURSE_HOSTNAME=www.example.com
+    - RUBY_GC_MALLOC_LIMIT=50000000
+  matrix:
+    - "RAILS_MASTER=0"
+    - "RAILS_MASTER=1"
+
+addons:
+  postgresql: 9.5
+  apt:
+    packages:
+    - gifsicle
+    - jpegoptim
+    - optipng
+    - jhead
+
+matrix:
+  allow_failures:
+    - env: "RAILS_MASTER=1"
+  fast_finish: true
+
+rvm:
+  - 2.4.1
+  - 2.3.3
+
+services:
+  - redis-server
+
+sudo: required
+dist: trusty
+cache: yarn
+
+before_install:
+  - git clone --depth=1 https://github.com/discourse/discourse.git build ; cd build ; rm -rf plugins/*
+  - gem install bundler
+  - git clone --depth=1 https://github.com/discourse/discourse-canned-replies.git plugins/discourse-canned-replies
+  - yarn global add eslint babel-eslint
+  - eslint --ext .es6 plugins/**/assets/javascripts
+
+before_script:
+  - bundle exec rake db:create db:migrate
+
+install:
+  - bash -c "if [ '$RAILS_MASTER' == '1' ]; then bundle update --retry=3 --jobs=3 arel rails seed-fu; fi"
```

Comment:
Sorry, I think I copy & pasted this from Discourse's .travis.yml without much thinking.